src/ol/renderer/webgl/PointsLayer.js

```diff
@@ -337,24 +348,54 @@ class WebGLPointsLayerRenderer extends LayerRenderer {
   }
 
   /**
-   * Will return the last shader compilation errors. If no error happened, will return null;
-   * @return {string|null} Errors, or null if last compilation was successful
-   * @api
-   */
-  getShaderCompileErrors() {
-    return this.helper_.getShaderCompileErrors();
-  }
-
-  /**
-   * Returns a texture of 1x1 pixel, white
+   * Rebuild internal webgl buffers based on current view extent; costly, should not be called too much
+   * @param {import("../../PluggableMap").FrameState} frameState Frame state.
    * @private
-   * @return {ImageData} Image data.
    */
-  getDefaultTexture() {
-    const canvas = document.createElement('canvas');
-    const image = canvas.getContext('2d').createImageData(1, 1);
-    image.data[0] = image.data[1] = image.data[2] = image.data[3] = 255;
-    return image;
+  rebuildBuffers_(frameState) {
+    const vectorLayer = /** @type {import("../../layer/Vector.js").default} */ (this.getLayer());
+    const vectorSource = vectorLayer.getSource();
+
+    this.verticesBuffer_.getArray().length = 0;
+    this.indicesBuffer_.getArray().length = 0;
+
+    // saves the projection transform for the current frame state
+    this.helper_.makeProjectionTransform(frameState, this.renderTransform_);
+    makeInverseTransform(this.invertRenderTransform_, this.renderTransform_);
+
+    // loop on features to fill the buffer
+    const features = vectorSource.getFeatures();
```

Comment:
I understand that this pull request is only a snapshot of a bigger refactoring. But just as note to self: here only features for the current extent and resolution should be fetched from the source.

Reply:
Sure, the logic here is very basic but requesting *all* features all the time has the advantage of giving a better user experience when zooming in/out. Obviously this would not work for even bigger datasets (although for them a tile based approach would make more sense).